File: swift/0001-two-sum.swift
// ! Day 1 // 9.28


class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var map: [Int:Int] = [:]
        for i in 0 ..< nums.count {
            if let mapped = map[target - nums[i]] {
                return [mapped, i]
            }
            map[nums[i]] = i 
        }
        return []
    }
}

File: javascript/0001-two-sum.js
// ! Day 1 // 9.28

var twoSum = function(nums, target) {
  const map = new Map();
  for (let i = 0; i < nums.length; i++) {
    let currVal = nums[i];
    if (map.has(currVal)) {
      return [map.get(currVal), i];
    }

    let diff = target - currVal;
    map.set(diff, i);
  }
};
